frontend/src/models/models.js

```diff
@@ -89,6 +89,37 @@ module.exports = app => app.component('models', {
     this.status = 'loaded';
   },
   methods: {
+    clickFilter(path) {
+      if (this.searchText) {
+        let searchObj;
+      
+        try {
+          // Try to parse this.searchText if it's a valid JSON string
+          searchObj = JSON.parse(this.searchText);
+        } catch (e) {
+          // If it fails, default to an empty object
+          console.log('could not add to filter', e);
+        }
+      
+        // Now assign the new property or update it
+        searchObj[path] = "";
+      
+        // Save it back to this.searchText as a valid JSON string
+        this.searchText = JSON.stringify(searchObj);
```

Comment:
Again, the priority is to avoid modifying the user's existing input as much as is reasonably possible. We don't want to replace the entirety of `searchText`